include/LightGBM/utils/common.h

```diff
@@ -1082,12 +1082,20 @@ struct __StringToTHelper<T, true> {
     // Fast (common) path: For numeric inputs in RFC 7159 format:
     const bool fast_parse_succeeded = fast_double_parser::parse_number(str.c_str(), &tmp);
 
-    // Rare path: Not in RFC 7159 format. Possible "inf", "nan", etc. Fallback to standard library:
+    // Rare path: Not in RFC 7159 format. Possible "inf", "nan", etc.
     if (!fast_parse_succeeded) {
-      std::stringstream ss;
-      Common::C_stringstream(ss);
-      ss << str;
-      ss >> tmp;
+      std::string strlower(str); 
```

Comment:
To pass linting tests
```
include/LightGBM/utils/common.h:1087:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
```
```suggestion
      std::string strlower(str);
```